Projects/UOContent/Network/Packets/IncomingPlayerPackets.cs

```diff
@@ -420,27 +424,36 @@ public static void DisplayGumpResponse(NetState state, SpanReader reader, int pa
                 return;
             }
 
-            var textEntries = new TextRelay[textCount];
+            TextRelay[] textEntries;
 
-            for (var i = 0; i < textEntries.Length; ++i)
+            if (textCount == 0)
             {
-                int entryID = reader.ReadUInt16();
-                int textLength = reader.ReadUInt16();
+                textEntries = [];
+            }
+            else
+            {
+                textEntries = new TextRelay[textCount];
 
-                if (textLength > 239)
+                for (var i = 0; i < textEntries.Length; ++i)
                 {
-                    state.LogInfo("Invalid gump response, disconnecting...");
-                    var exception = new InvalidGumpResponseException($"Text entry {i} is too long ({textLength})");
-                    exception.SetStackTrace(new StackTrace());
-                    NetState.TraceException(exception);
-                    state.Mobile?.SendMessage("Invalid gump response.");
+                    int entryID = reader.ReadUInt16();
+                    int textLength = reader.ReadUInt16();
 
-                    // state.Disconnect("Invalid gump response.");
-                    return;
-                }
+                    if (textLength > 239)
+                    {
+                        state.LogInfo("Invalid gump response, disconnecting...");
+                        var exception = new InvalidGumpResponseException($"Text entry {i} is too long ({textLength})");
+                        exception.SetStackTrace(new StackTrace());
+                        NetState.TraceException(exception);
+                        state.Mobile?.SendMessage("Invalid gump response.");
```

Comment:
We used to do this, but we don't want to make it easier for players to poke holes in badly created gumps, so we opt to disconnect them.